text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+- Feature Name: `target_tier_policy`
+- Start Date: 2019-09-20
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+We should have an official, objective policy for adding new (tier 3) targets,
+and for raising targets to tier 2 (with rustup builds) or even tier 1.
+
+# Motivation
+[motivation]: #motivation
+
+Rust developers regularly implement new targets in the Rust compiler, and
+reviewers of pull requests for such new targets would like a clear, consistent
+policy to cite for accepting or rejecting such targets. Currently, individual
+reviewers do not know what overall policy to apply, and whether to apply solely
+their own judgment or defer to a Rust governance team.
+
+Rust developers regularly ask how they can raise an existing target to tier 2
+(and in particular how they can make it available via rustup), and occasionally
+ask what it would take to add a new tier 1 target. The Rust project has no
+clear policy for target tiers. People not only don't know, they don't know who
+to ask or where to start.
+
+(See https://forge.rust-lang.org/platform-support.html for more information
+about targets and tiers.)
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+At a high level, the three tiers break down as follows:
+
+- Tier 3 targets provide no guarantees of support.
+- Tier 2 targets will always build, but may not pass tests.
+- Tier 1 targets will always build and pass tests.
```

Comment:
I think these should be inverted: Rust (the project & community) supports targets, not vice-versa. It would also be good to clarify that these guarantees are expected to hold for all "variants" of a particular `rustc` version, i.e., everything from the `master` branch builds to the stable channel. Finally, the tests in question should be made explicit (I assume this is referring to the test suite in `rust-lang/rust`?).

A possible rephrasing:

> At a high level, the guarantees provided by the Rust team for each tier are as follows:
>
> - Tier 3 targets: none.
> - Tier 2 targets: `rustc` can always be built on these targets, but the test suite may not pass.
> - Tier 1 targets: `rustc` can always be built on these targets, and its test suite must pass.

Reply:
Should be “`rustc` or `std` can always ...”

Reply:
I asked about `std` and `core` above; it also seems reasonable, I think, for some targets to support `alloc` but not all of `std`. 

I think this policy just needs to explicitly say something like "each target must define whether it supports all of `std`, just `alloc`, just `core`, or has no standard library support." (I originally thought this might be a missing "dimension" of the tier categorization, but since bare-metal targets can never support all of `std`,  bundling this information together with the "tier" seems unhelpful.)